lib/reporters/tap.js

```diff
@@ -27,6 +27,10 @@ TapReporter.prototype.start = function () {
 TapReporter.prototype.test = function (test) {
 	var output;
 
+	var passed = test.todo ? 'not ok' : 'ok';
+	var todoDirective = test.todo ? '# TODO' : '';
+	var skipDirective = test.skipped ? '# SKIP' : '';
```

Comment:
Perhaps do:

```
var directive = '';
if (test.todo) {
  directive = '# TODO';
} else if (test.skipped) {
  directive = '# SKIP';
}
```

Which simplifies the `todoDirective || skipDirective || ''` statement below and removes the unnecessary `|| ''` fall-through.